### PR TITLE
Back more metadata using per-query tables

### PR DESCRIPTION
**Pull request by cjgillot**
r? @ghost

### CODE REVIEW

**Comment by cjgillot** [comment]
r? rust-lang/compiler

**Comment by lcnr** [comment]
after a quick skim this lgtm

r? @petrochenkov

**Comment by petrochenkov** [comment]
@bors r+
This slightly increases max-rss though.

File: compiler/rustc_metadata/src/rmeta/mod.rs [hunk 1]
```diff
@@ -316,8 +329,8 @@ define_tables! {
 
 #[derive(Copy, Clone, MetadataEncodable, MetadataDecodable)]
 enum EntryKind {
-    AnonConst(mir::ConstQualifs, Lazy<RenderedConst>),
-    Const(mir::ConstQualifs, Lazy<RenderedConst>),
+    AnonConst,
+    Const,
```
Comment (eddyb) [inline compiler/rustc_metadata/src/rmeta/mod.rs:333]: Very nice! I wonder if some time soon this whole `enum` could be subsumed by just `DefKind`.

**Comment by cjgillot** [comment]
The perf results are overall green for rustdoc. Regression is localized to a single benchmark `externs` for debug/opt configurations, but not check. This regression is real, but on a stress test: loading `native_library_kind` is slower.
The size of on-disk metadata has grown: the encoding is less space-efficient than it used to be (5% larger for externs).

**Pull request merged**